crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -320,19 +330,12 @@ where
     fn submit_solution_response(&self, solution_response: SolutionResponse) -> RpcResult<()> {
         self.deny_unsafe.check_if_safe()?;
 
-        let solution_response_senders = self.solution_response_senders.clone();
-
-        // TODO: This doesn't track what client sent a solution, allowing some clients to send
-        //  multiple (https://github.com/paritytech/jsonrpsee/issues/452)
-
-        let mut solution_response_senders = solution_response_senders.lock();
-
         let slot = solution_response.slot_number;
+        let mut solution_response_senders = self.solution_response_senders.lock();
 
         let success = solution_response_senders
             .peek_mut(&slot)
-            .and_then(|senders| senders.pop())
-            .and_then(|mut sender| sender.send(solution_response).ok())
+            .and_then(|sender| sender.try_send(solution_response.solution).ok())
```

Comment:
Shouldn't we use `flatten` here?

Reply:
Use it how?

Reply:
Currently, `success` indicates that `peek_mut` returned `Some` rather than the result of `try_send`. If I'm not mistaken, we have `Some(Some(result))` and using `flatten` before` `is_some` should check the inner value.

Reply:
If we used `.map()` it would be `Option<Option<T>>`, but we use `.and_then()`, which gives us `Option<T>` without additional flattening required.

Reply:
Yep, playground confirmed that. My bad!